rfcs/min-relations.md

```diff
@@ -0,0 +1,479 @@
+                .checked_sub(amount_owed)
+                .expect("Nice kneecaps you got there...");
+            *money_query.get(lender).unwrap() += amount_owed;
+        }
+    }
+}
+```
+
+### Nuances of relations
+
+Relations between entities are automatically cleaned up when either the source or target entity is removed.
+Just like components, you can view which relations were removed in the last frame through the `RemovedRelation<T>` system parameter, which also returns their data.
+
+Each relation kind + target combination is treated like a unique type for the purposes of archetypes.
+This makes can be relevant to the performance of your game; accelerating target entity filtering but creating overhead if you have a very large number of distinct targets.
+As a result, changing the source or target of a relation can only be done while the app has exclusive access to the world, via `Commands` or in exclusive systems.
+Here's an example of how you might do so:
+
+```rust
+
+fn love_potion(
+    mut commands: Commands,
+    query: Query<Entity, &Relation<Loves>, With<Cute>>,
+    player_query: Query<Entity, With<Player>>){
+    
+    let player = player.single().unwrap();
+
+    for victim, former_love in query.iter(){
+
+        // Only one relation of a given kind can exist between a source and a target;
+        // like always, new relations overwrite existing relations
+        // with the same source, kind and target
+        for (_, former_lover) in former_love {
+            commands.entity(victim).change_target::<Loves>(former_lover, player); // This is unethical!!
+        }
+    }
+}
+
+fn adoption(
+    mut commands: Commands,
+    // As with components, you can query for relations that may or may not exist
+    query: Query<(Entity, &NewOwner, Option<&Relation<Owns>>), With<Kitten>>,
+) {
+   for (kitten, new_owner, previous_ownership) in query.iter(){
+       // Changing the target or the source will fail silently if no appropriate relation exists
+       match previous_ownership {
+           // We can change sources by controlling which entity owns the relation
+           // move_relation is directly analagous to move_component
+           Some(_, old_owner) => commands.entity(old_owner).move_relation::<Owns>(new_owner, kitten); // uwu :3
+           None => commands.entity(new_owner).insert_relation(Owns::default(), kitten); // uwu!!
+       }
+   } 
+}
+
+// `change_target` and `move_relation` preserve the relation's data
+// This system removes all springs attached to mass 1, and adds them to mass 2 instead
+fn reattach_springs(mut commands: Commands,
+                   selected_mass1: Res<FirstSelected>,
+                   selected_mass2: Res<SecondSelected>,
+                   query: Query<(Entity, Relation<Spring>)>
+                  ) {
+    
+    let m1 = selelected_mass1.entity;
+    let m2 = selelected_mass2.entity;
+
+    for (source_mass, spring) in query.iter_mut(){        
```

Comment:
this example could use `query.get_mut(source_mass)` to avoid the outside loop. We know which entity's relations we are interested in directly.